server/main.lua

```diff
@@ -20,31 +20,40 @@ local function checkItems(Player, items, needsAll, checkRemove)
 					if item.amount >= v then
 						if needsAll then
 							count += 1
-							if count == finalcount then
-								if checkRemove then
-									for i in pairs(items) do
-										item = Player.Functions.GetItemByName(i)
-										if item and Config.Consumables[i] then
-											Player.Functions.RemoveItem(i, item.amount >= Config.Consumables[i] and Config.Consumables[i] or 1)
-										end
-									end
-								end
-								return true
-							end
 						else
 							if checkRemove and Config.Consumables[item.name] then
 								Player.Functions.RemoveItem(item.name, item.amount >= Config.Consumables[item.name] and Config.Consumables[item.name] or 1)
 							end
 							return true
 						end
 					end
+					if needsAll then
+						if count == finalcount then
+							if checkRemove then
+								for i in pairs(items) do
+									item = Player.Functions.GetItemByName(i)
+									if item and Config.Consumables[i] then
+										Player.Functions.RemoveItem(i, item.amount >= Config.Consumables[i] and Config.Consumables[i] or 1)
+									end
+								end
+							end
+							return true
+						end
+					end
 				end
 			else
 				finalcount = #items
 				local item = Player.Functions.GetItemByName(v)
 				if item then
 					if needsAll then
 						count += 1
+					else
+						if checkRemove and Config.Consumables[item.name] then
+							Player.Functions.RemoveItem(item.name, item.amount >= Config.Consumables[item.name] and Config.Consumables[item.name] or 1)
+						end
+						return true
+					end
+					if needsAll then
```

Comment:
I don't see the point of this, this just seems like an extra if statement which could be just one as it was orginally